tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c

```diff
@@ -0,0 +1,302 @@
+ *
+ *  Expected behaviour:
+ *   - NULL pointer is returned as there is no room
+ */
+void test_full_list_all_keys_in_use(void)
+{
+	struct bt_keys *returned_key;
+	uint8_t id = BT_ADDR_ID_3;
+	bt_addr_le_t *addr = BT_ADDR_LE_3;
+
+	bt_unpair_fake.custom_fake = bt_unpair_unreachable_custom_fake;
+	bt_conn_foreach_fake.custom_fake = bt_conn_foreach_custom_fake_all_keys_in_use;
+
+	returned_key = bt_keys_get_addr(id, addr);
+
+	zassert_true(returned_key == NULL, "bt_keys_get_addr() returned a non-NULL reference");
+}
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - All Keys are not used with a connection
+ *   - CONFIG_BT_KEYS_OVERWRITE_OLDEST is enabled
+ *
+ *  Expected behaviour:
+ *   - Valid reference pointer is returned which matches the oldest
+ *     reference at index 1
+ */
+void test_full_list_no_keys_in_use(void)
+{
+	struct bt_keys *returned_key;
+	uint8_t id = BT_ADDR_ID_3;
+	bt_addr_le_t *addr = BT_ADDR_LE_3;
+	uint32_t expected_oldest_params_ref_idx;
+
+	expected_oldest_params_ref_idx = 0;
+	oldest_params_vector = &testing_id_addr_pair_lut[expected_oldest_params_ref_idx];
+	bt_unpair_fake.custom_fake = bt_unpair_custom_fake;
+	bt_conn_foreach_fake.custom_fake = bt_conn_foreach_custom_fake_no_keys_in_use;
+
+	returned_key = bt_keys_get_addr(id, addr);
+
+	zassert_true(returned_key != NULL, "bt_keys_get_addr() returned a NULL reference");
+	zassert_true(returned_key == returned_keys_refs[expected_oldest_params_ref_idx],
+		     "bt_keys_get_addr() returned reference doesn't match expected one");
+}
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - Oldest key at slot 0 is used with a connection
+ *   - CONFIG_BT_KEYS_OVERWRITE_OLDEST is enabled
+ *
+ *  Expected behaviour:
+ *   - Valid reference pointer is returned which matches the oldest
+ *     reference at index 1
+ */
+void test_full_list_key_0_in_use_key_1_oldest(void)
+{
+	struct bt_keys *returned_key;
+	uint8_t id = BT_ADDR_ID_4;
+	bt_addr_le_t *addr = BT_ADDR_LE_4;
+	uint32_t expected_oldest_params_ref_idx;
+
+	expected_oldest_params_ref_idx = 1;
+	oldest_params_vector = &testing_id_addr_pair_lut[expected_oldest_params_ref_idx];
+	bt_unpair_fake.custom_fake = bt_unpair_custom_fake;
+	bt_conn_foreach_fake.custom_fake = bt_conn_foreach_custom_fake_key_slot_0_in_use;
+
+	returned_key = bt_keys_get_addr(id, addr);
+
+	zassert_true(returned_key != NULL, "bt_keys_get_addr() returned a NULL reference");
+	zassert_true(returned_key == returned_keys_refs[expected_oldest_params_ref_idx],
+		     "bt_keys_get_addr() returned reference doesn't match expected one");
+}
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - Key at slot 0 is used with a connection
+ *   - Oldest key isn't used with a connection
```

Comment:
`oldest key (slot 2)` isn't used with a connection`
I don't really like this test case because it implicitly depends on the state set by the previous one (if I understood correctly).
Could you group the two into a single one?